File: Data Structures & Algorithms/climbing-stairs/submission-0.java
class Solution {
    public int climbStairs(int n) 
    {
        if(n == 1 || n == 2) return n;
        int n_1 = 1;
        int n_2 = 2;
        for(int i = 2;i<n;i++){
            int temp = n_1+n_2;
            n_1 = n_2;
            n_2 = temp;
        }
        return  n_2;
    }
}

File: Data Structures & Algorithms/task-scheduling/submission-0.py
import heapq
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        freq_map = [0]*26
        cycles = 0
        for task in tasks:
            freq_map[ord(task)-ord('A')] += 1
        pq = []
        print(freq_map)
        #max heap
        for task in freq_map:
            if task != 0:
                heapq.heappush(pq, -task)
        print(pq)
        while pq:
            temp = []
            for i in range(0,n+1):
                cycles += 1
                if pq:
                    task = -heapq.heappop(pq)
                    if task-1 != 0:
                        temp.append(-(task-1))
                if not pq and not temp:
                    return cycles
            for t in temp:
                heapq.heappush(pq, t)
            print(pq)
        return cycles
            

        